### ISSUE TITLE
unable to correctly resolve count ternary when providing global_network_id value

### ISSUE THREAD

**Issue by drewmullen** [aws-ia/terraform-aws-cloudwan#7]
```shell
$ cd examples/with_globalnetwork 
$ terraform apply
data.aws_networkmanager_core_network_policy_document.main: Reading...
data.aws_networkmanager_core_network_policy_document.main: Read complete after 0s [id=2948540126]
╷
│ Warning: Experimental feature "module_variable_optional_attrs" is active
│ 
│   on ../../providers.tf line 3, in terraform:
│    3:   experiments      = [module_variable_optional_attrs]
│ 
│ Experimental features are subject to breaking changes in future minor or patch releases, based on feedback.
│ 
│ If you have feedback on the design of this feature, please open a GitHub issue to discuss it.
╵
╷
│ Error: Invalid count argument
│ 
│   on ../../main.tf line 3, in resource "awscc_networkmanager_global_network" "global_network":
│    3:   count = var.global_network.id == null ? 1 : 0
│ 
│ The "count" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many instances will be created. To work around this, use the -target argument to first apply only the resources that the count depends on
```

**Comment by drewmullen** [comment]
tried `aws_networkmanager_global_network` also to see if the issue is provider related, same outcome

**Comment by drewmullen** [comment]
solved by #9
